sdks/java/core/src/main/java/org/apache/beam/sdk/util/FlexibleBackoff.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.client.util.BackOff;
+import org.joda.time.Duration;
+
+
+/**
+ * A {@link BackOff} for which the retry algorithm is customizable. By default the
+ * {@link FlexibleBackoff} uses exponential backoff (base exponent 1.5) with an initial backoff of
+ * 1 second. These parameters can be overridden with {@link FlexibleBackoff#withExponent(double)}
+ * and {@link FlexibleBackoff#withInitialBackoff(Duration)}, respectively, and the maximum backoff
+ * after exponential increase can be capped using {@link FlexibleBackoff#withMaxBackoff(Duration)}.
+ *
+ * <p>By default, and after modifying any of the above parameters, {@link FlexibleBackoff} does not
```

Comment:
This sounds weird -- each of the modifier methods should document what its default value was (in addition to the default creation method specifying that). Modifying one of the parameters shouldn't modify other parameters (eg., once set, the number of retries shouldn't go back to unlimitedwhen other things are modified).